src/Layer/LabelLayer.js

```diff
@@ -0,0 +1,221 @@
+import LayerUpdateState from 'Layer/LayerUpdateState';
+import ObjectRemovalHelper from 'Process/ObjectRemovalHelper';
+import Layer from 'Layer/Layer';
+import Coordinates from 'Core/Geographic/Coordinates';
+import Extent from 'Core/Geographic/Extent';
+import Label from 'Core/Label';
+import { FEATURE_TYPES } from 'Core/Feature';
+
+const coord = new Coordinates('EPSG:4326', 0, 0, 0);
+
+let content;
+const _extent = new Extent('EPSG:4326', 0, 0, 0, 0);
+
+/**
+ * A layer to handle a bunch of `Label`. This layer can be created on its own,
+ * but it is better to use the option `labelEnabled` on another `Layer` to let
+ * it work with it (see the `vector_tile_raster_2d` example).
+ *
+ * @property {boolean} isLabelLayer - Used to checkout whether this layer is a
+ * LabelLayer.  Default is true. You should not change this, as it is used
+ * internally for optimisation.
+ * @property {string} crs - The crs of this layer.
+ */
+class LabelLayer extends Layer {
+    constructor(id, crs) {
+        super(id);
+
+        this.isLabelLayer = true;
+        this.crs = crs;
+    }
+
+    /**
+     * Reads each {@link FeatureGeometry} that contains label configuration, and
+     * creates the corresponding {@link Label}. To create a `Label`, a geometry
+     * needs to have a `label` object with at least a few properties:
+     * - `content`, which refers to `Label#content`
+     * - `position`, which refers to `Label#position`
+     * - (optional) `config`, containing miscellaneous configuration for the
+     *   label
+     *
+     * The geometry (or its parent Feature) needs to have a Style set.
+     *
+     * @param {FeatureCollection} data - The FeatureCollection to read the
+     * labels from.
+     * @param {Extent} extent
+     *
+     * @return {Label[]} An array containing all the created labels.
+     */
+    convert(data, extent) {
+        const labels = [];
+
+        const layerField = this.style && this.style.text && this.style.text.field;
+
+        // Converting the extent now is faster for further operation
+        extent.as(data.crs, _extent);
+        coord.crs = data.crs;
+
+        data.features.forEach((f) => {
+            // TODO: add support for LINE and POLYGON
+            if (f.type !== FEATURE_TYPES.POINT) {
+                return;
+            }
+
+            const featureField = f.style && f.style.text.field;
+
+            f.geometry.forEach((g) => {
+                const minzoom = (g.properties.style && g.properties.style.zoom && g.properties.style.zoom.min)
```

Comment:
`Style.zoom == undefined` is possible?

Reply:
It can, but only for `this.style`. If you take a look at `View`, you have the following line:
```js
labelLayer.style = layer.style;
```
which default to `{}` if you check `ColorLayer`.